projects/packages/jetpack-mu-wpcom/src/features/wpcom-widgets/wpcom-widgets.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Widgets for WordPress.com
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * WP.com Widgets (in alphabetical order)
+ */
+require_once __DIR__ . '/class-jetpack-i-voted-widget.php';
+
+/**
+ * Tracks widget being added or deleted.
+ *
+ * @see https://wpcom.trac.automattic.com/changeset/17565
+ *
+ * @global array $wp_registered_widgets Registered widgets.
+ */
+function jetpack_mu_wpcom_ajax_save_widget_stats() {
+	if ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) {
+		return false;
+	}
+
+    // phpcs:disable WordPress.Security.NonceVerification.Missing
+	if ( ( function_exists( 'wpcom_is_vip' ) && wpcom_is_vip() ) || ! is_admin() || ! isset( $_POST['id_base'] ) ) {
+		return false;
+	}
+
+	$id_base       = sanitize_text_field( wp_unslash( $_POST['id_base'] ) );
+	$add_new       = isset( $_POST['add_new'] ) ? (bool) $_POST['add_new'] : false;
+	$delete_widget = isset( $_POST['delete_widget'] ) ? (bool) $_POST['delete_widget'] : false;
+
+	if ( $delete_widget && isset( $_POST['widget-id'] ) && isset( $GLOBALS['wp_registered_widgets'][ sanitize_text_field( wp_unslash( $_POST['widget-id'] ) ) ] ) ) {
+		do_action( 'jetpack_bump_stats_extras', 'widget_removed', $id_base );
+	} elseif ( $add_new ) {
+		do_action( 'jetpack_bump_stats_extras', 'widget_added', $id_base );
+	}
+    // phpcs:enable WordPress.Security.NonceVerification.Missing
+}
+add_action( 'widgets.php', 'jetpack_mu_wpcom_ajax_save_widget_stats' );
+
+/**
+ * Some widgets are getting long in the tooth, hardly used, or just not very useful.
+ * Disable these widgets on sites where they're not active.
+ */
+define(
+	'JETPACK_MU_WPCOM_RETIRED_WIDGETS',
+	array(
+		'i_voted' => 'Jetpack_I_Voted_Widget',
+	)
+);
+
+/**
+ * Unregister retired widgets that aren't currently active.
+ *
+ * @global array $retired_widgets List of retired widgets.
+ * @return void
+ */
+function jetpack_mu_wpcom_unregister_retired_widgets() {
+
+	if ( ( function_exists( 'wpcom_is_vip' ) && wpcom_is_vip() ) || ! is_admin() ) {
+		return;
+	}
+
+	// just in case this is in any way a saving operation
+	if ( ! isset( $_SERVER['REQUEST_METHOD'] ) || 'POST' === $_SERVER['REQUEST_METHOD'] ) {
+		return;
+	}
+
```

Comment:
The WPcom file doesn't remove these widgets if you're on a sandbox, but I think that's confusing because it makes it harder to debug fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qjvqtrgf.cuc%3Se%3Q2n18n44n%23156%2Q158-og

Also, this section only retires widgets for the legacy widget editor, which you can access by adding the disable widget block sticker.